crates/sui/src/client_commands.rs

```diff
@@ -470,6 +472,13 @@ impl SuiClientCommands {
                 let build_config =
                     resolve_lock_file_path(build_config, Some(package_path.clone()))?;
 
+                let lookup = resolve_manifest_lookup(Some(package_path.clone()))?;
```

Comment:
Note that when we are publishing the package, we want to get the published at addresses of all its transitive dependencies, rather than the package itself (which should not have a published-at address, because we are in the process of publishing it).  The resolution graph that we create before compiling will contain all the parsed manifest files (with all their custom properties), so I would consider trying to intercede between resolution and building to retain access to the resolution graph, and read from it directly, rather than try and re-open and re-parse all the manifest files.